framework/src/base/FEProblem.C

```diff
@@ -111,7 +111,7 @@ FEProblem::FEProblem(const InputParameters & parameters) :
     _dt(declareRestartableData<Real>("dt")),
     _dt_old(declareRestartableData<Real>("dt_old")),
 
-    _nl(getParam<bool>("use_nonlinear") ? *(new NonlinearSystem(*this, "nl0")) : *(new EigenSystem(*this, "nl0"))),
+    _nl(getParam<bool>("use_nonlinear") ? *(new NonlinearSystem(*this, "nl0")) : *(new MooseEigenSystem(*this, "eigen0"))),
```

Comment:
@YaqiWang @permcody  @fdkong 
this change breaks a capability that was only tested in MAMMOTH which we didn't see till I did a submodule update. We used to be able to restart a transient from an eigen checkpoint file. Now you get this
```
ERROR: no system named eigen0 found!
Stack frames: 12
0: 0   libmesh_opt.0.dylib                 0x0000000105695e39 libMesh::print_trace(std::__1::basic_ostream<char, std::__1::char_traits<char> >&) + 1337
1: 1   libmesh_opt.0.dylib                 0x0000000105693d39 libMesh::MacroFunctions::report_error(char const*, int, char const*, char const*) + 329
2: 2   libmesh_opt.0.dylib                 0x00000001056a34bb libMesh::System& libMesh::EquationSystems::get_system<libMesh::System>(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&) + 251
3: 3   libmesh_opt.0.dylib                 0x0000000105afdce4 void libMesh::EquationSystems::_read_impl<double>(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, libMesh::XdrMODE, unsigned int, bool) + 2628
4: 4   libmesh_opt.0.dylib                 0x0000000105afcfac void libMesh::EquationSystems::read<double>(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, libMesh::XdrMODE, unsigned int, bool) + 44
5: 5   libmoose-opt.0.dylib                0x0000000105079dd8 Resurrector::restartFromFile() + 168
6: 6   libmoose-opt.0.dylib                0x0000000104c2c510 FEProblem::initialSetup() + 416
7: 7   libmoose-opt.0.dylib                0x0000000104e162b8 Transient::init() + 1464
8: 8   libyak-opt.0.dylib                  0x0000000103fc7a8d OutputApp::executeExecutioner() + 221
9: 9   libmoose-opt.0.dylib                0x0000000104d4e8b6 MooseApp::run() + 118
10: 10  yak-opt                             0x0000000103d3f599 main + 105
11: 11  libdyld.dylib                       0x00007fff8c59e5c9 start + 1
[0] ./include/libmesh/equation_systems.h, line 689, compiled Sep 22 2016 at 10:03:51
```
which is caused because the underlying libmesh system now has a different name. 
Any suggestions?